ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -524,21 +535,36 @@ public short updateBlobExpiration(BlobId blobId, long expirationTime, CloudUpdat
     // It is absorbed by applyTtlUpdate::L1395.
     // The validator doesn't check for this, perhaps another gap in legacy code.
     if (isDeleted(cloudMetadata)) {
+      // Replication must first undelete the deleted blob, and then update-TTL.
       String error = String.format("Unable to update TTL of %s as it is marked for deletion in cloud", blobIdStr);
       logger.error(error);
       throw AzureCloudDestination.toCloudStorageException(error, new StoreException(error, StoreErrorCodes.ID_Deleted), azureMetrics);
     }
 
     try {
       if (!cloudUpdateValidator.validateUpdate(CloudBlobMetadata.fromMap(cloudMetadata), blobId, newMetadata)) {
-        // Legacy cloudBlobStore does not expect an exception. However,
-        // below is the correct behavior. For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
-        // We should never hit this case however because ReplicaThread::applyUpdatesToBlobInLocalStore does all checks.
-        // It is absorbed by applyTtlUpdate::L1395
-        azureMetrics.blobUpdateTTLErrorCount.inc();
+        /*
+          Legacy cloudBlobStore does not expect an exception. However, below is the correct behavior.
+          For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
+          ReplicaThread::applyUpdatesToBlobInLocalStore does all necessary checks before calling updateTTL however,
+          ReplicaThread::handleGetResponse does not.
+
+          We can come here from ReplicaThread::handleGetResponse L1258 that applies PUT+TTL without checking if the blob
+          uploaded is permanent. The validator reports the blob has already been ttl-updated, and we end up flooding the logs
+          and incrementing metrics. To avoid this, just don't inc any error metric or print any logs. However, this would
+          mask a scenario where we are erroneously trying to update the ttl of permanent blob. This is ok because we prevent
+          an unnecessary request to cloud without sacrificing any correctness. The ReplicaThread should actually check before
+          applying a ttl-update, but it was written for disk-based log, and not a cloud-based backup system.
```

Comment:
Since we are replicating from multiple replicas in separate threads, I am just thinking that this situation is unavoidable even if a replica thread checks the blob before applying a ttl-update.

Reply:
Is there a change you are asking me to make ? 

You are referring to a concurrent update which is possible but less probable (and harmless) than the PUT+TTL case mentioned here. PUT+TTL  _always_ falls in this branch. I have tested this for two weeks in ei. As mentioned, the fix should be in the repl-layer to not apply ttl-update if the expiry is already -1.